c2rust-analyze/src/rewrite/expr/mir_op.rs

```diff
@@ -360,20 +361,22 @@ impl<'a, 'tcx> ExprRewriteVisitor<'a, 'tcx> {
     fn visit_ptr_offset(&mut self, op: &Operand<'tcx>, result_ty: LTy<'tcx>) {
         // Compute the expected type for the argument, and emit a cast if needed.
         let result_ptr = result_ty.label;
-        let (result_own, result_qty) =
-            type_desc::perms_to_desc(self.perms[result_ptr], self.flags[result_ptr]);
-
-        let arg_expect_own = result_own;
-        // TODO: infer `arg_expect_qty` based on the type of offset this is (positive / unknown)
-        let arg_expect_qty = match result_qty {
-            Quantity::Single => Quantity::Slice,
-            Quantity::Slice => Quantity::Slice,
-            Quantity::OffsetPtr => Quantity::OffsetPtr,
+        let result_desc =
+            type_desc::perms_to_desc(result_ty.ty, self.perms[result_ptr], self.flags[result_ptr]);
+        let result_own = result_desc.own;
```

Comment:
i think it's worth destructuring `result_desc` rather than grabbing field by field

Reply:
I changed this and the other case below to just use `result_desc.own` instead of defining a separate `result_own` variable: 7384e347